src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/DeliverySessionController.kt

```diff
@@ -118,4 +133,88 @@ class DeliverySessionController(
     val user = userMapper.fromToken(authentication)
     return DeliverySessionDTO.from(deliverySessionService.submitSessionFeedback(actionPlanId, sessionNumber, user))
   }
+
+  @GetMapping("/referral/{referralId}/delivery-session-appointments/{appointmentId}")
+  fun getDeliverySessionAppointment(
+    @PathVariable(name = "referralId") referralId: UUID,
+    @PathVariable(name = "appointmentId") appointmentId: UUID,
+    authentication: JwtAuthenticationToken,
+  ): DeliverySessionAppointmentDTO {
+    val user = userMapper.fromToken(authentication)
+    val referral = referralService.getSentReferral(referralId) ?: throw ResponseStatusException(
+      HttpStatus.BAD_REQUEST, "sent referral not found [referralId=$referralId]"
+    )
+    referralAccessChecker.forUser(referral, user)
+    val matchingAppointment = deliverySessionService.getSessions(referralId)
```

Comment:
Is this not model/business related logic and hence reusable as a service method?

Reply:
In this case we are transforming the `Session`s' list of appointments into a list of appointment and sessionNumber pairs.

It's not possible to delegate this functionality into the service method since the sessionNumber is not retrievable from an `Appointment` object. We can't know which `Session` the retrieved `Appointment` belongs to without using some type of wrapper object; an `Appointment` has no concept of whether it's an initial assessment or a delivery session appointment.